pkg/controllers/deployment/deployment_controller.go

```diff
@@ -192,7 +196,17 @@ func (c *oauthServerDeploymentSyncer) Sync(ctx context.Context, syncContext fact
 	// deployment, have RV of all resources
 	expectedDeployment, err := getOAuthServerDeployment(operatorSpec, proxyConfig, c.bootstrapUserChangeRollOut, resourceVersions...)
 	if err != nil {
-		return nil, false, append(errs, err)
+		return nil, false, false, append(errs, err)
+	}
+
+	if oidcEnabled, err := c.oidcEnabled(); err != nil {
+		return nil, false, false, append(errs, err)
+	} else if oidcEnabled {
+		err := c.deployments.Deployments(expectedDeployment.Namespace).Delete(ctx, expectedDeployment.Name, metav1.DeleteOptions{})
+		if err != nil {
```

Comment:
I know this is a test PR, but we should strive to always handle IsNotFound errors.